sinzlab/datajoint-link
# Rework everything

- Update documentation
- Remove flagging-related logic
- Change commands of Activated -> Deprecated transition
- Transition flagged activated entity to deprecated
- Transition flagged received entity to tainted
- Remove update being able to have multiple commands
- Add layout section to docs
- Add persistence class diagram to docs
- Define interface of link gateway
- Define interface of DataJoint link facade
- Enable strict mypy checking for new facade module
- Rename some classes in docs
- Implement link creation from persistent data
- Fix description of deprecate activity
- Implement applying updates
- Implement fetch1 method
- Implement insert method
- Refactor gateway tests
- Add method to identification translator
- Use translator to translate multiple keys
- Split loop in link creating function
- Combine methods on datajoint facade
- Refactor method on datajoint gateway
- Extract method for updating a row
- Inline row variable
- Ask for confirmation when deleting
- Accept multiple updates
- Pass multiple keys to facade
- Extract function for getting rows in restriction
- Allow restriction with multiple keys
- Process multiple keys in facade
- Fix rows not being deleted in fake table
- Fix bug when updating rows in facade
- Rename method in facade
- Implement command grouping
- Add children to fake table
- Enable creation of child tables
- Convert state dict to dataclass
- Allow specification of children state
- Give fake tables a name
- Check child tables as well when checking state
- Allow setting state of child tables
- Add support for part tables
- Simplify application of updates
- Extract function for adding parts to local
- Remove unused fake table methods
- Simplify if condition in fake table
- Extract function for copying fake table
- Implement restricting with string in fake table
- Simplify fetching of tainted primary keys
- Implement external data handling in fake table
- Refactor fake table
- Allow external attrs when creating tables
- Fix bug in fake table copying
- Allow download path when checking state
- Test adding external data to local
- Test adding external file to local
- Remove external file stuff from original test
- Remove fake table test for external file
- Use temporary directory to download data
- Remove temp_path stuff
- Move persistence tests module to integration tests
- Add docstrings to DataJoint link gateway
- Add docstrings to datajoint link facade
- Fix gateway applying all updates to all keys
- Update modules checked by mypy
- Ensure data is fetched as dictionaries
- Use transactions in datajoint link facade
- Refactor deprecate command tests
- Add function for initializing tables and gateway
- Test transaction for more commands
- Test more commands regarding transaction safety
- Fix incorrect persistent update for finish delete
- Add rows to local table within transaction
- Move link gateway class
- Move link facade
- Update class diagram in docs
- Improve replace stores function
- Use table factories in facade
- Update packages
- Fix linter errors
- Add functions producing DJ infrastructure
- Add database server credentials providers
- Add definition providers
- Add outbound table name provider
- Remove static table definition provider
- Enable mypy check for possibly undefined variables
- Fix possibly undefined variable bug
- Add outbound schema name provider
- Move configuration code to new module
- Ignore undefined variables error from ruff
- Update type hint on lazy schema
- Remove tests for initialization of adapters
- Initialize link gateway
- Remove tests for use case initialization
- Fix incorrect outbound table definition
- Use strict checks on config module
- Check new use cases module with mypy
- Remove duplication from mypy configuration
- Ignore mypy error
- Always put restrictions in AndList
- Add type stubs for datajoint
- Pin numpy < 1.20
- Fix incorrect docstring in method of update
- Rename method on update
- Move testing related assignments module
- Add new pull and delete use-cases
- Return results from operations on entities/links
- Add response models to use-cases
- Add operation name to entity result
- Add operation name to link result
- Prohibit self transitions
- Remove request models
- Add datajoint controller
- Add datajoint configuration object
- Pass translator to function creating gateway
- Add datajoint presenter
- Add new local table mixin
- Replace gateway creating function
- Fix table load not returning instance
- Change how restrictions are passed to pull
- Fix mixin not fetching primary keys
- Add source property to mixin
- Add outbound property to mixin
- Replace old system with reworked one
- Remove obsolete framework code
- Remove obsolete adapter code
- Remove obsolete use-cases code
- Remove obsolete entities code
- Remove misc obsolete code
- Move top-level custom types module to adapters
- Remove adapters.datajoint package
- Remove frameworks.datajoint package
- Add type annotation to function
- Simplify mypy configuration
- Inline create table function
- Test pulling with external and part in single test
- Remover unnecessary pulling tests
- Move pulling test module on directory up
- Extract create table function
- Slide part table definition statement
- Pass parts as list
- Fix table formatting
- Fix incorrect deprecated state mapping
- Fix mro on local table class
- Fix accessing property on class instead of instance
- Convert table creating function to fixture
- Slide statements
- Rewrite table creating fixture
- Rewrite functional test for deletion
- Rename module containing deletion tests
- Explicitly specify outbound table on link creation
- Remove outbound table name provider
- Add another entity to functional pulling test
- Use restriction when pulling entities
- Move pull method to source table
- Fix deletion ignoring restriction
- Don't set local table as attribute on mixin
- Cache connections
- Remove pylint
- Remove lazy schema
- Move tests for docker runner to new package
- Move container runner module to tests
- Rename docker modules
- Remove unused fixtures
- Fix mypy errors in test modules
- Simplify mypy configuration
- Remove profiling stuff
- Remove entrypoints directory
- Remove functional tests for source attribute
- Add flagged method to source table
- Rename mixin to local mixin
- Add source table mixin
- Rename local table creating function
- Refactor mixins
- Update readme
